pkg/ignition/ignition.go

```diff
@@ -0,0 +1,118 @@
+package ignition
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ignCfg "github.com/coreos/ignition/v2/config/v3_2"
+	ignCfgTypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcfg "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups="machineconfiguration.openshift.io",resources=machineconfigs,verbs=list;watch
+
+const (
+	// kubeletSystemdName is the name of the systemd service that the kubelet runs under,
+	// this is used to parse the kubelet args
+	kubeletSystemdName = "kubelet.service"
+	// CloudConfigOption is the kubelet CLI option for the cloud configuration file
+	CloudConfigOption = "cloud-config"
+	// CloudProviderOption is the kubelet CLI option for cloud provider
+	CloudProviderOption = "cloud-provider"
+)
+
+var (
+	// renderedWorker holds the contents of the cluster's worker node ignition file
+	renderedWorker Data
+)
+
+// Data holds the fully contents of an ignition file
+type Data struct {
+	ignCfgTypes.Config
+	// KubeletUnit is the kubelet systemd service spec in ignition file
+	KubeletUnit ignCfgTypes.Unit
+}
+
+// Initialize populates the RenderedWorker global variable with the worker ignition file data.
+func Initialize(c client.Client) error {
+	log := ctrl.Log.WithName("ignition")
+
+	mcList := &mcfg.MachineConfigList{}
+	err := c.List(context.TODO(), mcList)
+	if err != nil {
+		return err
+	}
+	renderedWorkerPrefix := "rendered-worker-"
+	var rawIgnition []byte
+	for _, mc := range mcList.Items {
+		if strings.HasPrefix(mc.Name, renderedWorkerPrefix) {
```

Comment:
Is there a way to pick the latest? Or add a doc here saying why we are picking the first one we find.

Reply:
Picking the latest now